jobserver/api/jobs.py

```diff
@@ -195,6 +200,9 @@ def handle_alerts_and_notifications(request, job_request, job):
             user_id=job_request.created_by_id,
         )
 
+    if hasattr(job_request, "analysis_request"):
```

Comment:
Hmm, this seems odd, probably a previous decision, but could JobRequest.analysis_request be a property or similar, rather than a possibly missing attribute?

Reply:
This is an artefact of it being a OneToOneField defined on AnalysisRequest.  When the relationship is missing, the target model (JobRequest in this instance) doesn't have the instance variable defined.  It is _maddening_.